packages/http-api/src/abstract-http-client.ts

```diff
@@ -6,39 +6,45 @@ import {
     IHttpRequest,
     IHttpResponseMessage,
     IHttpRequestMessage,
-    IHttpResponseRejectMessage
+    IHttpResponseRejectMessage,
+    IQueue,
+    HttpClientParams,
 } from '@testring/types';
 import { loggerClient } from '@testring/logger';
 import { HttpCookieJar } from './cookie-jar';
+import { Queue } from '@testring/utils';
 
 const nanoid = require('nanoid');
 
 const toString = c => c.toString();
 
 export abstract class AbstractHttpClient implements IHttpClient {
     protected abstract broadcast(options: IHttpRequestMessage): void;
+    private queue: IQueue<Function>;
+    private queueRunning = false;
 
-    constructor(protected transportInstance: ITransport) {
+    constructor(protected transportInstance: ITransport, private params: HttpClientParams) {
+        this.queue = new Queue();
```

Comment:
Don't use common words for naming 
```suggestion
        this.requestQueue = new Queue();
```